### PR TITLE
Migrate hardcoded dependencies to version catalog

### PR DESCRIPTION
**Pull request by tuancoltech**
Migrate hardcoded dependencies to version catalog.

`[Gradle version catalogs](https://docs.gradle.org/current/userguide/platforms.html) enable you to add and maintain dependencies and plugins in a scalable way. Using Gradle version catalogs makes managing dependencies and plugins easier when you have [multiple modules](https://developer.android.com/topic/modularization). Instead of hardcoding dependency names and versions in individual build files and updating each entry whenever you need to upgrade a dependency, you can create a central version catalog of dependencies that various modules can reference in a type-safe way with Android Studio assistance.`

https://app.asana.com/0/1207819682524134/1208308884950165
https://developer.android.com/build/migrate-to-catalogs

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -80,34 +80,33 @@ android {
...
-    implementation 'androidx.preference:preference:1.2.0'
-    implementation "com.google.dagger:hilt-android:2.48"
-    kapt "com.google.dagger:hilt-compiler:2.48"
-    kapt 'androidx.hilt:hilt-compiler:1.0.0'
+    implementation(libraries.google.dagger.hilt.android)
+    kapt(libraries.google.dagger.hilt.compiler)
+    kapt(libraries.androidx.hilt.compiler)
 
-    implementation 'com.github.kirich1409:viewbindingpropertydelegate-noreflection:1.5.9'
+    implementation(libraries.viewbinding.property.delegate)
```
Comment (shubertm) [inline app/build.gradle:98]:
```suggestion
implementation(libraries.viewbinding.property.delegate.noreflection)
```

File: app/build.gradle [hunk 2]
```diff
@@ -80,34 +80,33 @@ android {
...
+    implementation(libraries.androidx.appcompat)
+    implementation(libraries.android.material)
+    implementation(libraries.androidx.constraintlayout)
+    implementation(libraries.androidx.navigation.fragment.ktx)
+    implementation(libraries.androidx.preference.ktx)
+    implementation(libraries.androidx.lifecycle.viewmodel.ktx)
 
-    implementation 'dev.arkbuilders:arkfilepicker:0.1.1'
-    implementation 'dev.arkbuilders:arklib:0.3.3'
+    implementation(libraries.arkbuilder.arkfilepicker)
```
Comment (shubertm) [inline app/build.gradle:91]:
```suggestion
implementation(libraries.arkbuilders.arkfilepicker)
```

File: app/build.gradle [hunk 3]
```diff
@@ -80,34 +80,33 @@ android {
...
+    implementation(libraries.arkbuilder.arklib)
```
Comment (shubertm) [inline app/build.gradle:92]:
```suggestion
implementation(libraries.arkbuilders.arklib)
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+google-firebase-crashlytics = { group = "com.google.firebase", name = "firebase-crashlytics", version = "none" }
+google-firebase-analytics = { group = "com.google.firebase", name = "firebase-analytics", version = "none" }
+
+android-material = { group = "com.google.android.material", name = "material", version.ref = "androidMaterial" }
+junit = { group = "junit", name = "junit", version.ref = "junit" }
+androidx-test-junit = { group = "androidx.test.ext", name = "junit", version.ref = "androidXTestJunit" }
+androidx-test-espresso = { group = "androidx.test.espresso", name = "espresso-core", version.ref = "androidXTestEspresso" }
+skydoves-balloon = { group = "com.github.skydoves", name = "balloon", version.ref = "skydovesBalloon" }
+flexbox = { group = "com.google.android.flexbox", name = "flexbox", version.ref = "flexbox" }
+arkbuilder-arkfilepicker = { group = "dev.arkbuilders", name = "arkfilepicker", version.ref = "arkFilepicker" }
```
Comment (shubertm) [inline gradle/libs.versions.toml:66]:
```suggestion
arkbuilders-arkfilepicker = { group = "dev.arkbuilders", name = "arkfilepicker", version.ref = "arkFilepicker" }
```

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+arkbuilder-arklib = { group = "dev.arkbuilders", name = "arklib", version.ref = "arkLib" }
```
Comment (shubertm) [inline gradle/libs.versions.toml:67]:
```suggestion
arkbuilders-arklib = { group = "dev.arkbuilders", name = "arklib", version.ref = "arkLib" }
```

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+[libraries]
+coil = { group = "io.coil-kt", name = "coil", version.ref = "coil" }
+coil-gif = { group = "io.coil-kt", name = "coil-gif", version.ref = "coil" }
+coil-svg = { group = "io.coil-kt", name = "coil-svg", version.ref = "coil" }
+coil-video = { group = "io.coil-kt", name = "coil-video", version.ref = "coil" }
+androidx-fragment-ktx = { group = "androidx.fragment", name = "fragment-ktx", version.ref = "androidxFragmentKtx" }
+fastadapter = { group = "com.mikepenz", name = "fastadapter", version.ref = "fastadapter" }
+fastadapter-extensions-binding = { group = "com.mikepenz", name = "fastadapter-extensions-binding", version.ref = "fastadapter" }
+fastadapter-extensions-diff = { group = "com.mikepenz", name = "fastadapter-extensions-diff", version.ref = "fastadapter" }
+arklib = { group = "dev.arkbuilders", name = "arklib", version.ref = "arkLib" }
```
Comment (shubertm) [inline gradle/libs.versions.toml:42]: This is redundant, should be removed

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+orbit-mvi-viewmodel = { group = "org.orbit-mvi", name = "orbit-viewmodel", version.ref = "orbitMvi" }
+orbit-mvi-compose = { group = "org.orbit-mvi", name = "orbit-compose", version.ref = "orbitMvi" }
+viewbinding-property-delegate = { group = "com.github.kirich1409", name = "viewbindingpropertydelegate-noreflection", version.ref = "viewbindingPropertyDelegate" }
```
Comment (shubertm) [inline gradle/libs.versions.toml:45]:
```suggestion
viewbinding-property-delegate-noreflection = { group = "com.github.kirich1409", name = "viewbindingpropertydelegate-noreflection", version.ref = "viewbindingPropertyDelegate" }
```

**Review by shubertm — CHANGES_REQUESTED** [review]

**Review by hieuwu — APPROVED** [review]

**Comment by tuancoltech** [comment]
@shubertm This PR's branch is managed in rebase-based strategy.
Please don't push merge commit to it.
I'll be hard-resetting it now.

**Comment by shubertm** [comment]
> @shubertm This PR's branch is managed in rebase-based strategy. Please don't push merge commit to it. I'll be hard-resetting it now.

So, would you please fix all issues I requested. Not all are fixed.

**Comment by tuancoltech** [comment]
> > @shubertm This PR's branch is managed in rebase-based strategy. Please don't push merge commit to it. I'll be hard-resetting it now.
> 
> So, would you please fix all issues I requested. Not all are fixed.

@shubertm You mean the changes you requested above?
I think all of them were fixed.
Could you do a git pull --rebase and check again?

**Comment by shubertm** [comment]
> > > @shubertm This PR's branch is managed in rebase-based strategy. Please don't push merge commit to it. I'll be hard-resetting it now.
> > 
> > 
> > So, would you please fix all issues I requested. Not all are fixed.
> 
> @shubertm You mean the changes you requested above? I think all of them were fixed. Could you do a git pull --rebase and check again?

Yes, the redundant dependency `arklib` is still there

**Comment by tuancoltech** [comment]
> > > > @shubertm This PR's branch is managed in rebase-based strategy. Please don't push merge commit to it. I'll be hard-resetting it now.
> > > 
> > > 
> > > So, would you please fix all issues I requested. Not all are fixed.
> > 
> > 
> > @shubertm You mean the changes you requested above? I think all of them were fixed. Could you do a git pull --rebase and check again?
> 
> Yes, the redundant dependency `arklib` is still there

@shubertm `arklib` is being used in `app/build.gradle` line 93.

**Comment by shubertm** [comment]
Yes, but in libs.versions.toml it is defined twice, line 39 and 64

**Comment by tuancoltech** [comment]
> Yes, but in `libs.versions.toml` it is defined twice, line 39 and 64

That's a good catch!
Removed it now.

**Review by hieuwu — APPROVED** [review]

**Pull request merged**